src/components/Popup.js

```diff
@@ -6,32 +6,41 @@ class Popup {
 
   open() {
     this._popupElement.classList.add("modal__open");
-    document.addEventListener("keydown", this._handleEscClose);
+    this.addEventListeners();
```

Comment:
Please, delete `this.addEventListeners();` from `open`. According to the task you need to have public method `setEventListeners` to call it once in `index.js`


otherwise you add listeners again and again to the same elements. This can cause a **`memory leak`**


Though you remove the listeners in `removeEventListeners` but... you don't remove the close icon listener and you don't correctly remove the submit handler in `PopupWithForm`